### PR TITLE
fix(epoch_manager): Some fixes and comments

### PR DESCRIPTION
**Pull request by mikhailOK**
Some fixes, refactoring and comments.

Test plan
---------
Run existing tests
Extra asserts in the code

### CODE REVIEW

File: chain/epoch_manager/src/lib.rs [hunk 1]
```diff
@@ -890,7 +905,9 @@ impl EpochManager {
         &mut self,
         block_info: &BlockInfo,
     ) -> Result<EpochId, EpochError> {
+        //        debug_assert_ne!(block_info.epoch_first_block, CryptoHash::default());
```
Comment (SkidanovAlex) [inline chain/epoch_manager/src/lib.rs]: ideally we should not have commented out code in the codebase

File: chain/epoch_manager/src/proposals.rs [hunk 2]
```diff
@@ -42,43 +41,64 @@ pub fn proposals_to_epoch_info(
...
+    // Account -> new_stake
     let mut stake_change = BTreeMap::new();
     let mut fishermen = vec![];
-    let mut fishermen_to_index = HashMap::new();
+    debug_assert!(
+        proposals.iter().map(|stake| &stake.account_id).collect::<HashSet<_>>().len()
+            == proposals.len(),
+        "Proposals should not have duplicates"
+    );
+    println!(
```
Comment (bowenwang1996) [inline chain/epoch_manager/src/proposals.rs]: remove?

File: chain/epoch_manager/src/lib.rs [hunk 3]
```diff
@@ -77,6 +77,8 @@ impl EpochManager {
                 validator_reward,
                 0,
             )?;
+            // HACK: dummy block info. This dummy is stored instead of genesis block at height = 0.
+            // Unclear how it works, but hard to change at this point.
```
Comment (bowenwang1996) [inline chain/epoch_manager/src/lib.rs]: why is it unclear how it works? Genesis block is treated specially anyways.

File: chain/epoch_manager/src/types.rs [hunk 4]
```diff
@@ -77,6 +77,7 @@ pub struct BlockInfo {
     pub epoch_id: EpochId,
     pub proposals: Vec<ValidatorStake>,
     pub chunk_mask: Vec<bool>,
+    /// Validators slashed since the start of epoch or in previous epoch ?
```
Comment (bowenwang1996) [inline chain/epoch_manager/src/types.rs]: Previous epoch

File: chain/epoch_manager/src/lib.rs (hunk 3 above)
Comment (mikhailOK) [reply chain/epoch_manager/src/lib.rs]: - dummy block is a parent of genesis block, but has the same height
- dummy block is its own parent
- dummy block and genesis block are both in epoch 0
- epoch 0 and epoch 1 have the same epoch id

very hard to reason about correctness of functions that don't necessarily have `if (close_to_genesis(block)) { special_treatment }`.

It seems like we can view this situation as dummy block being the start of the chain with `height = -1` and the height field is not accessed by anyone, but I'm not sure.

**Pull request merged**